airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/config_migrations.py

```diff
@@ -121,3 +121,78 @@ def transform(cls, config: Mapping[str, Any]) -> Mapping[str, Any]:
             config[stream_filter] = statuses
         # return transformed config
         return config
+
+
+class MigrateSecretsPathInConnector:
+    """
+    This class stands for migrating the config at runtime.
+    This migration is intended for backwards compatibility with the previous version, so existing secrets configurations gets migrated to new path.
+
+    Starting from `2.2.0`, the `client_id`, `client_secret` and `access_token` will be placed at `credentials` path.
+    """
+
+
+    @classmethod
+    def _should_migrate(cls, config: Mapping[str, Any]) -> bool:
+        """
+        This method determines whether the config should be migrated to nest existing fields at credentials.
+        It is assumed if credentials does not exist on configuration, `client_id`, `client_secret` and `access_token` exists on root path.
+        Returns:
+            > True, if the migration is necessary
+            > False, otherwise.
+        """
+        credentials = config.get("credentials", None)
+        return credentials is None or (
```

Comment:
I think this condition needs to change with the new release process. At first, we wanted to release both the platform and the source at the same time. This meant that we only had to bother about the migration once as:
* Before the update, access_token would be at the root but source would migrate this to `credentials.access_token`
* After the update:
    * when configuring the source, the access_token would be on `credentials.access_token`
    * when logging in through oauth, the platform would set the access_token would be on `credentials.access_token`
    
However, our new plan have those releases split. This means that we will have a state where the source is updated but not the platform. In this case, we could have the following steps:
* The source is updated
* The migration happens so `access_token` is moved to `credentials.access_token`
* A user does oauth, the access token is set to `access_token`
* Given this condition, there will be `credentials.access_token` and we won't leverage the new token from oauth

I think the condition here should just be "is there an access token on the root level". Does that make sense?